File: Signiture-Backend/kotlin/delivery/api/src/main/java/pcy/study/api/common/api/error/UserErrorCode.java
package pcy.study.api.common.api.error;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import pcy.study.api.common.api.ApiCode;

@Getter
@RequiredArgsConstructor
public enum UserErrorCode implements ApiCode {

    USER_NOT_FOUND(HttpStatus.BAD_REQUEST, 1404, "사용자를 찾을 수 없습니다."),
    ;

    private final HttpStatus httpStatus;
    private final Integer code;
    private final String message;
}

File: Signiture-Backend/kotlin/delivery/common/src/main/kotlin/pcy/study/common/api/code/UserErrorCode.kt
package pcy.study.common.api.code

import pcy.study.common.api.ApiCode

enum class UserErrorCode(
    private val httpStatusCode: Int,
    private val code: Int,
    private val message: String
) : ApiCode {

    USER_NOT_FOUND(400, 1404, "사용자를 찾을 수 없습니다."),
    ;

    override fun getHttpStatusCode(): Int {
        return this.httpStatusCode
    }

    override fun getCode(): Int {
        return this.code
    }

    override fun getMessage(): String {
        return message
    }
}